ddtrace/internal/processor/trace.py

```diff
@@ -141,6 +141,9 @@ def on_span_finish(self, span):
                 else:
                     finished = trace_spans
 
+                if self._partial_flush_enabled and trace.num_finished >= self._partial_flush_min_spans:
```

Comment:
we should probably do this check only once

```python
should_partial_flush = self._partial_flush_enabled and trace.num_finished >= self._partial_flush_min_spans
if trace.num_finished == len(trace.spans) or should_partial_flush:
    # do stuff

    if should_partial_flush:
        log.debug(num_finished)
```

Reply:
what you haven't isn't too bad performance wise, just thinking about legibility and in the future if we need to adjust the logic for "should partial flush" we can do so in only one place.